### ISSUE TITLE
Can FlushTimeout be configured per environment?

### ISSUE THREAD

**Issue by neilmulhy** [NLog/NLog#3707]
NLog.Web.AspNetCore: 4.9.0
NLog: 4.6.8
Asp.Net Core v3.0

I have a BufferWrapper and want to have different timeouts dependant upon which environment I'm in (Dev / Staging / Prod).

I thought I could achieve this by using the ${configsetting} layout renderer but the FlushTimeout attribute doesn't seem to support reading from a config file?

**EDIT:** I guess the answer to the above question is no because it accepts an Integer rather than a Layout?

``` xml
<target xsi:type="BufferingWrapper"
          name="String"
          bufferSize="Integer"
          flushTimeout="Integer" />
```

Any ideas how I can achieve this? Ideally, I'd want a small flush time in dev and staging but a longer time in production.

In earlier versions of nlog, I could've had an nlog.config file per environment using the below in startup:

``` c#
loggerFactory.AddNLog();
env.ConfigureNLog($"nlog.{env.EnvironmentName}.config");
```

But the startup code in v3.0 has changed so the nlog.config gets read in the program.cs, unless there's an environment override method that I've missed in the docs?

``` c#
var logger = NLog.Web.NLogBuilder.ConfigureNLog("nlog.config").GetCurrentClassLogger();
```

**Comment by snakefoot** [comment]
I guess it is a duplicate of #1234

While waiting this to be resolved then you could try and use environment specific NLog.config-files.

**NLog.config**
```xml
<nlog>
    <variables>
        <variable name="flushTimeout" value="30000" /> 
    <variables>
    <include file="Nlog-common.config" />
</nlog>
```

**NLog-common.config**
```xml
<nlog>
      <targets>
           <target type="BufferingWrapper" name="buffer" flushTimeout="${flushTimeout}">
                <target type="Console" name="console" />
           </target>
      </targets>
      <rules>
           <logger name="*" writeTo="buffer" />
      </rules>
</nlog>
```

See also: https://github.com/nlog/nlog/wiki/XML-config-include-example

**Comment by snakefoot** [comment]
I guess it is a duplicate of #1234

While waiting this to be resolved then you could try and use environment specific NLog.config-files.

**NLog.config**:
```xml
<nlog>
    <variables>
        <variable name="flushTimeout" value="30000" /> 
    <variables>
    <include file="Nlog-common.config" />
</nlog>
```

**NLog-common.config**
```xml
<nlog>
      <targets>
           <target type="BufferingWrapper" name="buffer" flushTimeout="${flushTimeout}">
                <target type="Console" name="console" />
           </target>
      </targets>
      <rules>
           <logger name="*" writeTo="buffer" />
      </rules>
</nlog>
```

See also: https://github.com/nlog/nlog/wiki/XML-config-include-example

**Comment by neilmulhy** [comment]
Thanks again @snakefoot for your great suggestions.

I ended up going for the variable override approach which came with a couple of unexpected challenges which I'll share for others.

**My implementation**

I had to create a file with the extension of .cfg as it turns out that targets automatically publish *.config fiiles by [default](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/visual-studio-publish-profiles?view=aspnetcore-3.0#exclude-files). I was getting a duplicate file warning and I didn't want to change default behaviour so went for having a different extension.

I ended up calling the file nlog.nonprod.cfg.

I have my nlog.config in a config folder. In order to ensure that the new cfg file wasn't copied to production I added the 2 item groups to the end of my web.csproj which did some conditional copying and not copying the new file to the output folder based on whether the configuration was for "Release" or not.

``` xml

  <ItemGroup Condition="'$(Configuration)' != 'Release'">
    <None Update="Config\nlog.nonprod.cfg">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
  </ItemGroup>

  <ItemGroup Condition="'$(Configuration)' == 'Release'">
    <None Update="Config\nlog.nonprod.cfg">
      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
    </None>
  </ItemGroup>
```
I found I needed to add the two in the order I did to get the publish behaviour I wanted.

It probably would've been more straight forward setting the new file to never be copied and putting an exclude for the file on the production publish profile but hey ho, it's working.
